src/OutputReaders/field_time_series_indexing.jl

```diff
@@ -227,6 +229,18 @@ end
 ##### FieldTimeSeries updating
 #####
 
+# Let's make sure `times` is available on the CPU. This is always the case 
+# for ranges. if `times` is a vector that resides on the GPU, it has to be moved to the CPU  for safe indexing. 
+# Copying the whole array is a bit unclean but considering `update_field_time_series!` is called quite rarely 
+# it should not impact too much memory / performance
+cpu_safe_interpolating_time_indices(time_indexing, times, t, arch)  = interpolating_time_indices(time_indexing, times, t)
```

Comment:
```suggestion
cpu_safe_interpolating_time_indices(arch, times, time_indexing, times, t)  = interpolating_time_indices(time_indexing, times, t)
```

I think this is a better ordering indicating the relative importance of dispatch on the different types.

Also isn't this function "gpu safe?" The issue is when `times` is on GPU. When it's on the CPU there's no need for a special method. You may want to rename it to `gpu_safe_interpolating_time_indices`

Reply:
the issue is when you are interpolating on the CPU indices on the GPU. So the interpolation should be CPU-available

Reply:
Still confused. Do we simply need to delete times = on_architecture(architecture(grid), times) in the constructor?

Reply:
When `times` is not a range, we need it as a `CuArray` in the kernels (somewhat equivalently to a coordinate like `λᶠᵃᵃ` for grids). What breaks the code is trying to index `times` on the CPU while interpolating